appdynamics/provider.go

```diff
@@ -21,13 +22,32 @@ func Provider() *schema.Provider {
 			"appdynamics_appd_service":     dataSourceAppdService(),
 		},
 		Schema: map[string]*schema.Schema{
-			"secret":              {Type: schema.TypeString, Sensitive: true, Required: true},
+			"secret":              {Type: schema.TypeString, Sensitive: true, Optional: true},
```

Comment:
Not sure if both secret and client_secret set to optional is a good idea. Maybe additional field defining secret type?

Reply:
Secret is to keep backward compatibility